src/main/resources/assets/js/components/AppSearchComponent.jsx

```diff
@@ -0,0 +1,52 @@
+/** jsx React.DOM */
+
+var React = require("react/addons");
+
+var AppSearchComponent = React.createClass({
+  displayName: "AppSearchComponent",
+
+  propTypes: {
+    onSearch: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      searchValue: null
+    };
+  },
+
+  handleChange: function (event) {
+    this.setState({searchValue: event.target.value});
```

Comment:
You can access the value through a React reference here, like:
```
this.refs.searchInput.value
```
This ref has to be defined on the input element below.